packages/zoltan2/src/input/Zoltan2_Adapter.hpp

```diff
@@ -146,9 +156,9 @@ template <typename User>
   // */ 
   virtual void getWeightsView(const scalar_t *&wgt, int &stride,
                               int idx = 0) const {
-    Kokkos::View<scalar_t *> tempWeightsView;
-    getWeightsKokkosView(tempWeightsView, idx);
-    wgt = tempWeightsView.data();
+    Kokkos::View<scalar_t *, typename node_t::device_type> kokkos_wgts;
```

Comment:
MJ will use the ids only for migration. So we had decided to make ids host only since otherwise we would have a copy to host, MPI calls, then a copy back to device.